docker/Dockerfile.worker-gpu

```diff
@@ -0,0 +1,31 @@
+FROM ufoym/deepo:all-py36
```

Comment:
I might be mistaken but I can't find a difference between this image and `docker/Dockerfile.worker` that you've provided above. Is there a reason we need two?

Reply:
IMHO the base image is an ugly piece of a dockerfile. It has a single `RUN` statement to run all the package installation commands, be it `apt-get`, `pip install`, `git clone`... for each and every software artifact that is installed in the container... (see: https://github.com/ufoym/deepo/blob/master/docker/Dockerfile.all-py27) this approach fundamentally neglects the philospohy of layered dockerfiles... thus everytime a `docker build` fails because of any small issue during software installation, everything that has been done before in the build, is thrown away when a build fails... also it never removes unnecessary temporary data like `rm -rf .git` after every `git clone` or `rm -rf .cache` after every `pip install`... thus this dockerfile becomes unnecessarily bloated...

Reply:
I remove the build on top of  deepo. As recommended by @ianlivingstone, i build the main image on top of nvidia/cuda then add python3.6 and run pip to install other requirements. I did not use the already provided ubuntu installation script are they are installing ipfs which will be useless here.

I still need two images cpu/gpu because installing tensorflow using pip install tensorflow does not allow tensorflow to actually detect the gpus. You can check by running 

```
import tensorflow as tf
sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))
```
So we need one image, where tensorflow is installed for cpu usage and one in tensorflow-gpu for those who want to use gpus. Ideally, compose does not build anything and just use **openmined/grid** and **openmined/grid:gpu** where the second  is built using a Dockerfile that looks like 

```
FROM openmined/grid

RUN pip install tensorflow-gpu
```

Using compose, you cannot really build one Dockerfile on top of another, so they are still very similiar Dockerfile/Dockerfile.gpu where the only difference is the install of tensorflow_gpu for the second one..

Pytorch is much better on that side, just takes whatever it finds :)

Anyway, I test both cpu/gpu version and it seems to work.